lwr:
  # CONTROLLERS USED IN THE EXAMPLE
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 100  

  # THIS CONFIGURATION IS ACTUALLY SENDING TAU = K*Q
  joint_trajectory_controller:
    type: position_controllers/JointTrajectoryController
    joints: 
      - lwr_a1_joint
      - lwr_a2_joint
      - lwr_e1_joint
      - lwr_a3_joint
      - lwr_a4_joint
      - lwr_a5_joint
      - lwr_a6_joint

## OTHER CUSTOM CONTROLLERS LEFT HERE AS EXAMPLES
  # FtSensor Controller
  ft_sensor_controller:
    type: lwr_force_position_controllers/FtSensorController
    root_name: vito_anchor
    tip_name: lwr_7_link
    topic_name: /lwr/ft_sensor
    ft_sensor_offset: [0, 0, 0, 0, 0, 0]
    base_tool_weight_com: [0, 0, 0, 0, 0, 0]
    p_wrist_toolcom: [0, 0, 0.1]
    publish_rate: 200 #Hz

  # Cartesian Position Controller
  cartesian_position_controller:
    type: lwr_force_position_controllers/CartesianPositionController
    root_name: vito_anchor
    tip_name: lwr_7_link
    p_wrist_ee: [0, 0, 0.025]

  # Hybrid Impedance Controller 
  # (Spong, Journal of Robotics And Automation, Vol. 4, Issue 5, 1988)
  hybrid_impedance_controller:
    type: lwr_force_position_controllers/HybridImpedanceController
    root_name: vito_anchor
    tip_name: lwr_7_link
    internal_motion_controlled_link: lwr_3_link
    p_wrist_ee: [0, 0, 0.025]
    p_base_ws: [-0.7, 0, 0.05]                                                                                                                
    ws_base_angles: [0, 0, 0]
    publish_rate: 1 #Hz